Corona-Warn-App/src/test/java/de/rki/coronawarnapp/submission/ExposureKeyHistoryCalculationsTest.kt

```diff
@@ -83,18 +77,122 @@ class ExposureKeyHistoryCalculationsTest {
 
     @Test
     fun test_toExternalFormat() {
+        // regular case
+        var tek1 = createKey(thisMorning)
+        var tek2 = createKey(thisMorning.minusDays(1))
+        var result = instance.toExternalFormat(
+            listOf(tek1, tek2),
+            TransmissionRiskVector(intArrayOf(0, 1, 2)),
+            intArrayOf(3998, 3999, 4000),
+            thisMorning
+        )
         Assert.assertArrayEquals(
-            intArrayOf(10, 20), instance.toExternalFormat(
-                listOf(
-                    createKey(0),
-                    createKey(1)
-                ),
-                TransmissionRiskVector(intArrayOf(0, 1, 2))
-            ).map { it.rollingStartIntervalNumber }.toTypedArray().toIntArray()
+            intArrayOf(tek1.rollingStartIntervalNumber, tek2.rollingStartIntervalNumber),
+            result.map { it.rollingStartIntervalNumber }.toTypedArray().toIntArray()
+        )
+        Assert.assertArrayEquals(
+            intArrayOf(3998, 3999),
+            result.map { it.daysSinceOnsetOfSymptoms }.toTypedArray().toIntArray()
+        )
+        Assert.assertArrayEquals(
+            intArrayOf(0, 1),
+            result.map { it.transmissionRiskLevel }.toTypedArray().toIntArray()
+        )
+
+        // gap
+        tek1 = createKey(thisEvening)
+        tek2 = createKey(thisEvening.minusDays(7))
+        result = instance.toExternalFormat(
+            listOf(tek1, tek2),
+            TransmissionRiskVector(intArrayOf(0, 1, 2, 3, 4, 5, 6, 7)),
+            intArrayOf(3998, 3999, 4000, 4001, 4002, 4003, 4004, 4005),
+            thisEvening
+        )
+        Assert.assertArrayEquals(
+            intArrayOf(tek1.rollingStartIntervalNumber, tek2.rollingStartIntervalNumber),
+            result.map { it.rollingStartIntervalNumber }.toTypedArray().toIntArray()
+        )
+        Assert.assertArrayEquals(
+            intArrayOf(3998, 4005),
+            result.map { it.daysSinceOnsetOfSymptoms }.toTypedArray().toIntArray()
+        )
+        Assert.assertArrayEquals(
+            intArrayOf(0, 7),
+            result.map { it.transmissionRiskLevel }.toTypedArray().toIntArray()
+        )
```

Comment:
Recommendation: might be worth refactoring into three local helper functions for assertions; each function accepting `result` and an array with either the expected `rollingStartIntervalNumber`, the DSOS, or the TRL.